src/ajax/xhr.js

```diff
@@ -19,7 +19,19 @@ jQuery.ajaxSettings.xhr = window.ActiveXObject !== undefined ?
 			// and http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9
 			// Although this check for six methods instead of eight
 			// since IE also does not support "trace" and "connect"
-			/^(get|post|head|put|delete|options)$/i.test( this.type ) &&
+			//
+			// Support: IE 10-11
+			// IE seems to error on cross-domain PATCH requests when ActiveX XHR
+			// is used. Use the following strategy:
+			// 1. If request is local, use the ActiveX XHR.
+			// 2. Otherwise, in IE9+ always use the native XHR.
+			// 3. Otherwise, in IE8 if request type is *not* included in the following
+			//    list, use the ActiveX XHR.
+			// 4. Otherwise, use the native XHR.
+			// jshint -W018
+			( !( document.documentMode < 9 ) ||
+			// jshint +W018
+				/^(get|post|head|put|delete|options)$/i.test( this.type ) ) &&
```

Comment:
> That seems more readable but what's the point of the last ` || createActiveXHR()` in your example?

Because `createStandardXHR` has a `try` block and can return `undefined` when "[Enable native XMLHTTP support](https://msdn.microsoft.com/en-us/library/ms537505(v=vs.85).aspx)" is disabled. That's also the reason why this expression wasn't _already_ using a ternary conditional.